src/com/magento/idea/magento2plugin/actions/generation/generator/CronjobClassGenerator.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+package com.magento.idea.magento2plugin.actions.generation.generator;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiDirectory;
+import com.intellij.psi.PsiFile;
+import com.jetbrains.php.lang.psi.PhpFile;
+import com.jetbrains.php.lang.psi.elements.PhpClass;
+import com.magento.idea.magento2plugin.actions.generation.data.CronjobClassData;
+import com.magento.idea.magento2plugin.actions.generation.generator.util.DirectoryGenerator;
+import com.magento.idea.magento2plugin.actions.generation.generator.util.FileFromTemplateGenerator;
+import com.magento.idea.magento2plugin.bundles.ValidatorBundle;
+import com.magento.idea.magento2plugin.indexes.ModuleIndex;
+import com.magento.idea.magento2plugin.magento.files.CronjobTemplate;
+import com.magento.idea.magento2plugin.util.GetPhpClassByFQN;
+import org.jetbrains.annotations.NotNull;
+
+import javax.swing.*;
+import java.io.File;
+import java.util.Properties;
+
+public class CronjobClassGenerator extends FileGenerator {
+    private CronjobClassData cronjobClassData;
+    private Project project;
+    private ValidatorBundle validatorBundle;
+    private final DirectoryGenerator directoryGenerator;
+    private final FileFromTemplateGenerator fileFromTemplateGenerator;
+
+    /**
+     * @param project
+     * @param cronjobClassData
+     */
+    public CronjobClassGenerator(Project project, @NotNull CronjobClassData cronjobClassData) {
+        super(project);
+        this.project = project;
+        this.cronjobClassData = cronjobClassData;
+
+        this.directoryGenerator = DirectoryGenerator.getInstance();
+        this.fileFromTemplateGenerator = FileFromTemplateGenerator.getInstance(project);
+        this.validatorBundle = new ValidatorBundle();
+    }
+
+    /**
+     *
+     * @param actionName
+     *
+     * @return void
+     */
+    public PsiFile generate(String actionName) {
+        String errorTitle = "Error";
+//        String cronjobFqn = this.getCronjobFqn();
```

Comment:
Probably you forget some to remove this :)

Reply:
Refactored ✅